### PR TITLE
fixed tests for link to external pages challenge

### LINKED ISSUE #41173: Link to External Pages with Anchor Elements test is too permissive
**Describe your problem and how to reproduce it:**
the challenge can be passed with an url like `href="https://freecatphotoapp.com."`, or `href="https://freecatphotoapp.com/stuuyuuufushtfgh"`

**Add a Link to the page with the problem:**
https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/link-to-external-pages-with-anchor-elements

**Suggestion:**
```diff
-assert(/https:\/\/(www\.)?freecatphotoapp\.com/gi.test($('a').attr('href')));
+assert(/^https:\/\/(www\.)?freecatphotoapp\.com$/i.test($('a').attr('href')));
```

 - add anchors to make sure that the link is exactly that with no extra stuff around
 - remove the `g` tag which in this case is just useless
 - ~[possibly] add the possibility to have an `http` link pass, seeing that the video hint show that link (it shows an example `https`, and then it deletes the `s`)~

### PR DESCRIPTION
**Pull request by majestic-owl448**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #41173 

I've made the discussed changes:
 - added anchors to the regex to make sure nothing around the required url is allowed
 - removed the `g` because useless
 - allowed both `http` and `https` as per in the discussion in the issue

### CODE REVIEW

**Review by ShaunSHamilton — DISMISSED** [review]
This LGTM 👍 

---

Closes the issue, but...
Some irony: We are preventing the tests from being too permissive, but also allowing something _incorrect_ to pass...

**Review by RandellDawson — CHANGES_REQUESTED** [review]
See my suggested change and question.

File: curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/link-to-external-pages-with-anchor-elements.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ assert(/cat photos/gi.test($('a').text()));
 You need an `a` element that links to `https://freecatphotoapp.com`
 
 ```js
-assert(/https:\/\/(www\.)?freecatphotoapp\.com/gi.test($('a').attr('href')));
+assert(/^https?:\/\/(www\.)?freecatphotoapp\.com$/i.test($('a').attr('href')));
```
[the remark this answers is not in the document]
Comment (nhcarrigan) [reply curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/link-to-external-pages-with-anchor-elements.md:35]: Agreed - I'd be in favour of disallowing `www`.

[the remark this answers is not in the document]
Comment (ieahleen) [reply curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/link-to-external-pages-with-anchor-elements.md:35]: Should I change that too?

**Comment by ieahleen** [comment]
should I remove the `www` from the regex? I asked it in the comment discussion but as I committed , it was dismissed

**Comment by RandellDawson** [comment]
@ieahleen I think not allowing the `www` is what we want.

**Review by RandellDawson — APPROVED** [review]

**Comment by RandellDawson** [comment]
Can we have another mod review this PR?

**Review by ShaunSHamilton — APPROVED** [review]
This LGTM 👍

**Pull request merged**